sharedb-codemirror.js

```diff
@@ -199,7 +199,7 @@ ShareDBCodeMirror.prototype.applyOp = function(op, source) {
     return;
   }
 
-  if (source === CODE_MIRROR_OP_SOURCE) {
+  if (source === true || source === CODE_MIRROR_OP_SOURCE) {
```

Comment:
It looks like in ShareDB 1.0.0 beta, they [replaced the `source` argument to `doc.submitOp` with an `options` object](https://github.com/share/sharedb/commit/fc58bc768ed52591a312e30f4da792f2bbc48931). I think that's what caused the `source` value here to change.

Based on ShareDB's [sharedb-string-binding](https://github.com/share/sharedb-string-binding), it looks like the correct behavior now is to [pass `{source: this}` to `doc.submitOp`](https://github.com/share/sharedb-string-binding/blob/master/index.js#L95) and [check for `source === this` when receiving an op](https://github.com/share/sharedb-string-binding/blob/master/index.js#L51).

So I think we should change the `submitOp` call in `onOp` above from:
```
shareDoc.submitOp(docOp, source);
```
to
```
shareDoc.submitOp(docOp, {source: this});
```
And change the check here to:
```
if (source === this)
```

And of course at that point, `CODE_MIRROR_OP_SOURCE` can be removed.

Reply:
I edited this by referencing line number 741 of https://github.com/share/sharedb/blob/master/lib/client/doc.js.
If you have any idea about name of variables in test/mocks/mock-doc.js , please tell me or change by yourself.